### PR TITLE
Add deep nesting support for .get(), .set() and .del()

### PR DESCRIPTION
**Pull request by morkro**
Here is the pull request for #33 .
I've added tests, updated the README a little and bumped the version.

### CODE REVIEW

**Comment by rexxars** [comment]
`del` doesn't work as expected here:

```js
conf.set('foo.bar.baz', {awesome: true});
conf.set('foo.bar.bang', {awesome: false});

conf.del('foo.bar.baz');
console.log(this.conf.get('foo.bar.bang'));
// Expected: {awesome: false}
// Actual: undefined
```

**Comment by rexxars** [comment]
Also, the changes done in README stripped significant whitespace (two spaces + linebreak === hard break) - so it's probably best if you put those back :-)

Otherwise, looks good!

**Comment by morkro** [comment]
Thanks, I didn't realise I removed the breaks!

I couldn't reproduce your example though. It works for me. Might be that the result is `undefined` because you're missing `this` in your example: `conf.set('foo.bar.baz', {awesome: true});`

It should be `this.conf.set('foo.bar.baz', {awesome: true});`.

**Comment by morkro** [comment]
Okay, the breaks are somehow removed by my IDE. I added `<br>` to make them appear again. Hope that's fine.

**Comment by rexxars** [comment]
The following should reproduce the issue - paste it into the `del()` test:

```js
this.conf.set('foo.bar.baz', {awesome: 'icecream'});
this.conf.set('foo.bar.zoo', {awesome: 'redpanda'});

this.conf.del('foo.bar.baz');
assert.equal(this.conf.get('foo.bar.zoo.awesome'), 'redpanda');
```

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "configstore",
-  "version": "1.4.0",
+  "version": "2.0.0",
```
Comment (sindresorhus) [inline package.json]: Don't bump this.

Comment (morkro) [inline package.json]: Sure, will revert that. Thought I could bump it because you were talking about a major update. Sorry for that :grimacing:

Comment (sindresorhus) [inline package.json]: Yes, but I have a script for that. And I might want to do other changes before doing a release.

**Comment by morkro** [comment]
Updated to previous feedback. 
- `.del()` now uses the `.delete()` method from `dot-prop`
- reverted the version back to `1.4.0`

Hope this PR is finally working for everyone. Sorry for the issues.

**Comment by sindresorhus** [comment]
Looks good. Thanks :)

**Pull request closed without merging**